File: src/java/Models/Livre.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Models;

/**
 *
 * @author Thierry Christophe
 */
public class Livre {

   
    private int idlivre=0;
    
    private String design="l";
    private String codelivre="";
    private int exemplaire=0;
    
    /**
     * @return the idlivre
     */
    public Livre(){
        
    }
    public Livre(int id,String c, String d,int e){
        idlivre=id;
        codelivre=c;
        design=d;
        exemplaire=e;
    }
     public Livre( String d,int e){
        design=d;
        exemplaire=e;
    }
    public int getIdLivre() {
        return idlivre;
    }

    /**
     * @param idlivre the idlivre to set
     */
    public void setIdLivre(int idlivre) {
        this.idlivre = idlivre;
    }
 /**
     * @return the codelivre
     */
    public String getCodeLivre() {
        return codelivre;
    }

    /**
     * @param codelivre the codelivre to set
     */
    public void setCodeLivre(String codelivre) {
        this.codelivre = codelivre;
    }
    
    /**
     * @return the design
     */
    public String getDesign() {
        return design;
    }

    /**
     * @param design the design to set
     */
    public void setDesign(String design) {
        this.design = design;
    }

    /**
     * @return the exemplaire
     */
    public int getExemplaire() {
        return exemplaire;
    }

    /**
     * @param exemplaire the exemplaire to set
     */
    public void setExemplaire(int exemplaire) {
        this.exemplaire = exemplaire;
    }
    
    
}

File: web/js/api/Livre.js
/* 
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/JSP_Servlet/JavaScript.js to edit this template
 */
function getLivre(){
    let tbody=document.getElementById('employeeTableBody')
    api.get('Livre/get').then((r)=>{
        console.log("succes")
        tbody.innerHTML=r
    }).catch((console.error()))
}

async function addLivre(d){
    const r = await api.post("Livre/add",d)
    console.log(r)
}
async function updateLivre(d){
    const r = await api.post("Livre/update",d);
    console.log(r);
}

function deleteLivre(id){
    const formData = new URLSearchParams();
    console.log(id)
    formData.append("idlivre", id);
    api.post('Livre/delete',formData).then((r)=>{
        console.log(r+'ff')
        if(r=="success")getLivre()
    }).catch((e)=>{ console.error()})
}
function setDataUpdate(idLivre,design,exemplaire){
    let idLivreInput=document.getElementById('idLivreUpdate')
    let designInput=document.getElementById('designUpdate')
    let exemplaireInput=document.getElementById('exemplaireUpdate')

    idLivreInput.value=idLivre
    designInput.value =design
    exemplaireInput.value=parseInt(exemplaire)    
    console.log(idLivreInput)
    
}